HealthSDK/GiniHealthSDK/Sources/GiniHealthSDK/Core/PaymentComponent/PaymentComponentView.swift

```diff
@@ -165,105 +172,92 @@ final class PaymentComponentView: UIView {
     
     private func setupView() {
         self.translatesAutoresizingMaskIntoConstraints = false
-        self.frame = CGRect(x: 0, y: 0, width: .max, height: 240)
-        
+        self.frame = CGRect(x: 0, y: 0, width: .greatestFiniteMagnitude, height: Constants.viewHeight)
+
         self.backgroundColor = viewModel.backgroundColor
         
         self.addSubview(contentStackView)
-        
-        // Content StackView Constraints
-        NSLayoutConstraint.activate([
-            heightAnchor.constraint(equalToConstant: frame.height),
-            contentStackView.leadingAnchor.constraint(equalTo: leadingAnchor),
-            contentStackView.trailingAnchor.constraint(equalTo: trailingAnchor),
-            contentStackView.topAnchor.constraint(equalTo: topAnchor, constant: 16),
-            contentStackView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: 4)
-        ])
 
         contentStackView.addArrangedSubview(moreInformationStackView)
         contentStackView.addArrangedSubview(selectBankView)
-        setupMoreInformationView()
-        setupSelectBankView()
-        setupBankPickerView()
-        setupPayInvoiceView()
-    }
-    
-    private func setupMoreInformationView() {
-        // We need our label into a view for layout purposes. Stackviews require views in order to satisfy all dynamic constraints
-        let moreInformationLabelView = UIView()
         moreInformationLabelView.addSubview(moreInformationLabel)
-        NSLayoutConstraint.activate([
-            moreInformationLabel.leadingAnchor.constraint(equalTo: moreInformationLabelView.leadingAnchor),
-            moreInformationLabel.trailingAnchor.constraint(equalTo: moreInformationLabelView.trailingAnchor),
-            moreInformationLabel.centerYAnchor.constraint(equalTo: moreInformationLabelView.centerYAnchor)
-        ])
         moreInformationStackView.addArrangedSubview(moreInformationLabelView)
         moreInformationStackView.addArrangedSubview(moreInformationButton)
-    }
-    
-    private func setupSelectBankView() {
-        selectBankView.addSubview(selectBankLabel)
+        selectBankView.addSubview(selectYourBankLabel)
         selectBankView.addSubview(selectBankPickerView)
         selectBankView.addSubview(payInvoiceView)
         selectBankView.addSubview(poweredByGiniView)
-        
+        if viewModel.isPaymentProviderInstalled {
+            selectBankPickerView.addSubview(bankImageView)
+        }
+        selectBankPickerView.addSubview(bankNameLabel)
+        selectBankPickerView.addSubview(chevronDownIconView)
+        payInvoiceView.addSubview(payInvoiceLabel)
+
+        activateConstraints()
+    }
+
+    private func activateConstraints() {
```

Comment:
[Clarification]: @razvancapra can we refactor it even in more granular methods separating by view for example, so far for this method we have 46 lines 😅?